cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -743,23 +744,116 @@ public void updateSchedule(HttpRequest request, HttpResponder responder,
     doUpdateSchedule(request, responder, namespaceId, appName, appVersion, scheduleName);
   }
 
-  // TODO: refactor doAddSchedule and doUpdateSchedule to reduce code duplication
   private void doUpdateSchedule(HttpRequest request, HttpResponder responder, String namespaceId, String appId,
```

Comment:
I remembered we talked about getting rid of the update schedule API since it is hard to define the right semantic (e.g. the trigger/constraint changed while the schedule has a job in the job queue). Do we want to keep the API now?